### ISSUE TITLE
Deepmask does not detecting the whole object, mask only covers a part of the object after training with custom dataset

### ISSUE THREAD

**Issue by gnc10** [facebookresearch/deepmask#84]
I have trained deepmask with custom dataset containing 1000 images with just one category. The object proposal mask given by deepmask does not cover the whole object as shown in the attached images.

![res28](https://cloud.githubusercontent.com/assets/22654477/22008996/f19e8d90-dca5-11e6-85ee-ef9082ae7aa9.jpg)
![res](https://cloud.githubusercontent.com/assets/22654477/22009077/b0a88c18-dca6-11e6-9bdf-81d1aff2f8ca.jpg)

These two images are from deepmask trained with custom dataset. As you can see the car is not being detected fully. Also these two images were present in training dataset. Can anyone please let me know why the object is not being detected correctly by the deepmask? Am I missing something?

Also the output during training deepmask are as follows:-

[train] | epoch 00194 | s/batch 0.46 | loss: 0.03252 
[train] | epoch 00195 | s/batch 0.46 | loss: 0.03207 
[test]  | epoch 00195 | IoU: mean 020.72 median 018.31 suc@.5 004.02 suc@.7 002.14 | acc 098.49 | bestmodel x
[train] | epoch 00196 | s/batch 0.46 | loss: 0.03124 
[train] | epoch 00197 | s/batch 0.46 | loss: 0.03139 
[test]  | epoch 00197 | IoU: mean 021.72 median 019.61 suc@.5 003.62 suc@.7 002.26 | acc 098.16 | bestmodel x
[train] | epoch 00198 | s/batch 0.46 | loss: 0.03257 
[train] | epoch 00199 | s/batch 0.46 | loss: 0.03159 
[test]  | epoch 00199 | IoU: mean 022.69 median 020.77 suc@.5 003.67 suc@.7 002.18 | acc 098.23 | bestmodel x
[train] | epoch 00200 | s/batch 0.46 | loss: 0.03264 
[train] | epoch 00201 | s/batch 0.46 | loss: 0.03166 
[test]  | epoch 00201 | IoU: mean 023.17 median 021.16 suc@.5 004.02 suc@.7 001.92 | acc 097.93 | bestmodel x
[train] | epoch 00202 | s/batch 0.46 | loss: 0.03215 

Any help would be much appreciated. Thanks in advance!

**Comment by toomanyqs** [comment]
I am trying to segment a single large object from an image and facing the same issue. The object nearly covers 70% of the area in the images. What changes do I do to achieve satisfactory result ?

**Comment by sebaschaal** [comment]
Have you tried to experiment with the parameters initial scale, final scale, scale step and number of proposals?
I am using Deepmask/Sharpmask to detect objects close together in a box and found the ideal parameter setting for those case. I know 4 objects are in the box and 0.5 is the right scale, so I just use that one.

E.g. run demo with:
th computeProposals.lua $DEEPMASK/pretrained/sharpmask/ -si 0.5 -sf 0.5 -np 4 -img $DEEPMASK/data/test.jpg

**Comment by AyeshaGurnani** [comment]
Hello, what is the format of using custom dataset. Can you tell me once. What is the format? And what are the commands for using Custom Dataset?
